conans/client/installer.py

```diff
@@ -672,6 +672,10 @@ def _call_package_info(self, conanfile, package_folder, ref, is_editable):
 
                     conanfile.package_info()
 
+                    # this is necessary to check if the components declared for legacy generators
+                    # have the same namespace as the root cpp_info
+                    conanfile.cpp_info._raise_incorrect_components_names()
```

Comment:
This should not be necessary for 2.0, with CMakeDeps being free to specify any target name for components without restrictions.